assets/app/view/game/part/assignments.rb

```diff
@@ -34,29 +34,84 @@ def preferred_render_locations
 
         def load_from_tile
           @assignments = @tile.hex&.assignments || {}
+          @assignments_to_show = []
+          stack_group_count = {}
```

Comment:
Setting a default value of zero for the hash would simplify your code later on.
```suggestion
          stack_group_count = Hash.new(0)
```